### DISCUSSION TITLE
Add support for tab completion

### DISCUSSION THREAD

**Discussion by InvalidJoker** [pelican-dev/panel#785] [Ideas]
Since Pelican uses xterm, a POSIX-compliant terminal, it could support a real TTY to enable features like tab completion, syntax highlighting, and more.

**Comment by notAreYouScared** [comment]
I'm not sure that's going to be possible... as the console data is received over a websocket and we don't connect to the docker terminal for the container directly.

**Reply by InvalidJoker** [reply]
> I'm not sure that's going to be possible... as the console data is received over a websocket and we don't connect to the docker terminal for the container directly.

I think it is if we send everything through the web socket and use a terminal like xterm on the frontend which is POSIX-compliant it would work

**Reply by notAreYouScared** [reply]
I have no interest in attempting to figure this out...
